exe/CMakeLists.txt

```diff
@@ -13,7 +13,15 @@ include_directories(
     ${LEATHERMAN_INCLUDE_DIRS}
 )
 
-add_executable(puppet-db ${PROJECT_NAME}.cc)
+add_executable(puppet-query puppet-query.cc)
+target_link_libraries(puppet-query
+    lib${PROJECT_NAME}
+    ${Boost_PROGRAM_OPTIONS_LIBRARY}
+    ${Boost_FILESYSTEM_LIBRARY}
+    ${LEATHERMAN_LIBRARIES}
+)
+
```

Comment:
It's not critical for now, and I don't know what's appropriate with cmake, but in the make/automake world, I'd probably consider putting these common bits into a shared variable or similar, i.e. so you'd have target_link_libraries(... ${pdb_bin_libs}) or whatever.